velox/functions/lib/window/tests/WindowTestBase.cpp

```diff
@@ -48,6 +49,80 @@ WindowTestBase::QueryInfo WindowTestBase::buildWindowQuery(
   return {op, functionSql, querySql};
 }
 
+namespace {
+void buildOrderByParameters(
+    const std::string& overClause,
+    vector_size_t startIdx,
+    vector_size_t length,
+    std::vector<std::string>& orderByClauses) {
+  auto orderByPars = overClause.substr(startIdx, length);
+  std::istringstream tokenStream(orderByPars);
+  std::string token;
+  while (std::getline(tokenStream, token, ',')) {
+    orderByClauses.push_back(token);
+  }
+}
+}; // namespace
+
+WindowTestBase::QueryInfo WindowTestBase::buildStreamingWindowQuery(
+    const std::vector<RowVectorPtr>& input,
+    const std::string& function,
+    const std::string& overClause,
+    const std::string& frameClause) {
+  std::string functionSql =
+      fmt::format("{} over ({} {})", function, overClause, frameClause);
+  core::PlanNodePtr op = nullptr;
+  std::vector<std::string> orderByClauses;
+
+  std::string partitionByStr = "partition by";
+  auto partitionByStartIdx = overClause.find(partitionByStr);
+
+  std::string orderByStr = "order by";
+  auto orderByStartIdx = overClause.find(orderByStr);
+
+  // Extract the partition by keys.
+  if (partitionByStartIdx != std::string::npos) {
+    auto startIdx = partitionByStartIdx + partitionByStr.length();
+    auto length = 0;
+    if (orderByStartIdx != std::string::npos) {
+      length = orderByStartIdx - startIdx;
+    } else {
+      length = overClause.length() - startIdx;
+    }
+    buildOrderByParameters(overClause, startIdx, length, orderByClauses);
+  }
+  // Add NULLS FIRST after partition by keys. Because the
+  // defaultPartitionSortOrder in Window.cpp is NULLS FIRST.
+  for (auto i = 0; i < orderByClauses.size(); i++) {
+    if (orderByClauses[i].find("NULLS") == std::string::npos &&
+        orderByClauses[i].find("nulls") == std::string::npos) {
+      orderByClauses[i] = orderByClauses[i] + " NULLS FIRST";
+    }
+  }
+
+  // Extract the order by keys.
+  if (orderByStartIdx != std::string::npos) {
+    auto startIdx = orderByStartIdx + orderByStr.length();
+    auto length = overClause.length() - startIdx;
+    buildOrderByParameters(overClause, startIdx, length, orderByClauses);
```

Comment:
Is the NULLS fixing as done on line 94 not needed here ?

Reply:
No. The [defaultPartitionSortOrder ](https://github.com/facebookincubator/velox/blob/6fd816db78a9d7a53a76ad0ec76318cd3c05d919/velox/exec/WindowBuild.cpp#L40)only used in partition keys not order by keys.